incubator/hnc/pkg/forest/forest.go

```diff
@@ -317,11 +317,26 @@ func (ns *Namespace) HasCondition() bool {
 	return len(ns.conditions) > 0
 }
 
+// HasLocalCritCondition returns if the namespace has any local critical condition.
+func (ns *Namespace) HasLocalCritCondition() bool {
+	return ns.GetCondition(Local) != nil
+}
+
+// hasCritAncestorCondition returns if the namespace has CritAncestor condition.
+func (ns *Namespace) hasCritAncestorCondition() bool {
+	ans := ns.Parent()
+	for ans != nil {
+		if ans.HasLocalCritCondition() {
+			return true
+		}
+		ans = ans.Parent()
+	}
+	return false
+}
+
 // HasCritCondition returns if the namespace has any critical condition.
 func (ns *Namespace) HasCritCondition() bool {
-	// For now, all the critical conditions are set locally. It may not be true for
-	// future critical conditions. We may not want to just use local conditions.
-	return ns.GetCondition(Local) != nil
+	return ns.HasLocalCritCondition() || ns.hasCritAncestorCondition()
 }
```

Comment:
Perhaps this can be written more concisely as:

```
func (ns *Namespace) HasCritCondition() bool {
  if ns.HasLocalCritCondition() {
    return true
  }
  if ns.Parent() == nil {
    return false
  }
  return ns.Parent().HasCritCondition()
}
```